ci: Added pre-commit-hook

Sloox opened the pull request:
Fixes #1305

Needs to be tested locally to actually see a change. 
It requires a once off execution of the link Hook sh file.
Simply try git commit and it should run detekt.

jan-gogo reviewed (changes requested):
The linkHooks.sh was not executable so I have changed it using `chmod +x`.
After that, I have added empty lines to random kotlin file and tested pre-commit
```
MacBook-Pro-jan-gogo:flank-review jan-gogo$ git commit -m 'Test pre-commit'
hint: The '.githooks/pre-commit' hook was ignored because it's not set as executable.
hint: You can disable this warning with `git config advice.ignoredHook false`.
[#1305-git-hooks 19bd8833] Test pre-commit
 1 file changed, 9 insertions(+)
```
So I have changed `pre-commit` to executable and tried test again.
This is what I get:
```
MacBook-Pro-jan-gogo:flank-review jan-gogo$ git commit -m 'Test pre-commit'
fatal: cannot run .githooks//pre-commit: No such file or directory
```
Strange

Sloox commented:
> The `linkHooks.sh` was not executable so I have changed it using `chmod +x`.
> After that, I have added empty lines to a random kotlin file and tested pre-commit
> 
> ```
> MacBook-Pro-jan-gogo:flank-review jan-gogo$ git commit -m 'Test pre-commit'
> hint: The '.githooks//pre-commit' hook was ignored because it's not set as executable.
> hint: You can disable this warning with `git config advice.ignoredHook false`.
> [#1305-git-hooks 19bd8833] Test pre-commit
>  1 file changed, 9 insertions(+)
> ```
> 
> So I have changed `pre-commit` to executable and tried test again.
> This is what I get:
> 
> ```
> MacBook-Pro-jan-gogo:flank-review jan-gogo$ git commit -m 'Test pre-commit'
> fatal: cannot run .githooks//pre-commit: No such file or directory
> ```
> 
> Strange

Please retest it was a faulty shebang line.

adamfilipow92 commented:
Maybe we could add information about run ```linkHooks.sh``` before first commit to documentation in ```Environment setup``` section? It could help when for all new contributors. WDYT?

piotradamczyk5 commented:
One small issue on Mac
```
.githooks/linkHooks.sh: line 2: @echo: command not found
Changing git config for hooks to .githooks
Complete!
```

piotradamczyk5 commented:
> One small issue on Mac
> 
> ```
> .githooks/linkHooks.sh: line 2: @echo: command not found
> Changing git config for hooks to .githooks
> Complete!
> ```

I fixed it

piotradamczyk5 commented on .githooks/linkHooks.sh:
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+set echo of
```
```suggestion
set echo off
```

piotradamczyk5 commented:
> set echo off?

fixed

jan-gogo reviewed (approved):
I am not sure if `pre-commit` script will work on windows so `linkHooks.bat` probably is not necessary, but also I am not sure if any dev will use windows as the main os for development, so it's not a problem.

Sloox commented:
> I am not sure if `pre-commit` script will work on windows so `linkHooks.bat` probably is not necessary, but also I am not sure if any dev will use windows as the main os for development, so it's not a problem.

I use windows and it works correctly 👍
